File: js/day22.js
const readline = require('readline');

const r1 = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

r1.question(`Enter a number between 0 and 100: `, (answer) => {
  try {
    const num = Number(answer);

    // Validate if it is a number and not an empty string
    if (answer.trim() === '' || isNaN(answer)){
      throw new Error('That is not a valid number.');
    }

    // Check range boundaries
    if (num < 0 || num > 100){
      throw new Error('Out of range! The number must be between 0 and 100.');
    }

    console.log(`Success! You entered: ${num}`);
  } catch (err) {
    console.log(`Error: ${err.message}`);
  } finally {
    r1.close();
  }
});

File: python/day22.py
try:
  # Ask for input and convert it to a number
  num = float(input("Enter a number between 0 and 100: "))

  # Check the range
  if num < 0 or num > 100:
    raise ValueError(f"Input {num} is out of range! Must be between 0 and 100.")
    print(f"Success! You entered: {num}")

except ValueError as e:
  # This catches both non-numeric inputs and our custom range error
  print(f"Error: {e}")
